s3-backup.py

```
@@ -1,27 +1,94 @@
-import rclone
+import boto3
 import os
 import shutil
 from datetime import datetime
 
 
 def main():
+    directory_path = "/tmp/backup"
 
-    cfg_path = r'.rclone.conf'
+    archive_name = 'pelican-local-env-s3-' + datetime.strftime(datetime.utcnow(), "%Y-%m-%dT%H-%M-%S") + '.backup'
+    
+    s3_download = boto3.client(
+        's3',
+        aws_access_key_id=os.getenv('AWS_ACCESS_KEY_ID_DOWNLOAD'),
+        aws_secret_access_key=os.getenv('AWS_SECRET_ACCESS_KEY_DOWNLOAD'),
+        endpoint_url=os.getenv('AWS_HOST_DOWNLOAD'),
+    )
+    
+    s3_upload = boto3.client(
+        's3',
+        aws_access_key_id=os.getenv('AWS_ACCESS_KEY_ID_UPLOAD'),
+        aws_secret_access_key=os.getenv('AWS_SECRET_ACCESS_KEY_UPLOAD'),
+        endpoint_url=os.getenv('AWS_HOST_UPLOAD'),
+    )
 
-    with open(cfg_path) as f:
-        cfg = f.read()
+    bucket_name_download = os.getenv('AWS_BUCKET_NAME_DOWNLOAD')
+    bucket_name_upload = os.getenv('AWS_BUCKET_NAME_UPLOAD')
+    
+    download_dir(directory_path, bucket_name_download, s3_download)
 
-    archive_name = "s3-backup" + '_' + datetime.strftime(datetime.utcnow(), "%Y.%m.%d.%H-%M-%S") + 'UTC' + '.backup'
+    if os.path.exists(directory_path):
```

if without else, need to throw a meaningful exception and fail fast